doc/user_guide/deployment.rst

```diff
@@ -0,0 +1,156 @@
+MCP Server Deployment
+=====================
+
+The MCP server can be deployed either locally, as a Python package, or as a remote HTTP
+server. This section of the guide explains the details of how to run the MCP server as
+an HTTP server.
+
+Direct Python Deployment
+------------------------
+
+Install the Exasol MCP Server Python package. Below are examples of `pip` and `uv` commands.
+
+.. code-block:: shell
+
+   pip install exasol-mcp-server
+
+.. code-block:: shell
+
+   uv tool install exasol-mcp-server@latest
+
+Set the required environment variables and, optionally, the settings json file as described
+in the :doc:`server_setup` and other guides referenced there.
+
+Start the HTTP server from the command line, as in the example below.
+
+.. code-block:: shell
+
+    exasol-mcp-server-http --port <server-port>
+
+Docker Deployment
+-----------------
+
+When running the MCP Server from a Docker container, the environment variables and the settings
+file shall be passed as the arguments to the `docker run` command, as in the example below.
+
+Create a file `settings.json` and save it somewhere in the host machine.
+
+.. code-block:: json
+
+    {
+        "enable_read_query": true
+    }
+
+Now create and run the Docker container.
+
+.. code-block:: shell
+
+    docker run \
+        -p 7766:4896 \
+        -e EXA_DSN=my_dsn \
+        -e EXA_USER=my_user-name \
+        -e EXA_PASSWORD=my-password \
+        -e FASTMCP_SERVER_AUTH=exa.fastmcp.server.auth.oauth_proxy.OAuthProxy \
+        -e EXA_AUTH_UPSTREAM_AUTHORIZATION_ENDPOINT=my_identity_provider/oauth2/authorize \
+        -e EXA_AUTH_UPSTREAM_TOKEN_ENDPOINT=my_identity_provider/oauth2/token \
+        -e EXA_AUTH_UPSTREAM_CLIENT_ID=my_client_id \
+        -e EXA_AUTH_UPSTREAM_CLIENT_SECRET=my_client_secret \
+        -e EXA_AUTH_JWKS_URI=my_identity_provider/jwks \
+        -e EXA_AUTH_BASE_URL=my_mcp_server \
+        -e EXA_MCP_SETTINGS=/app/settings.json \
+        -v local_path_to_settings.json:/app/settings.json \
+        exadockerci4/exasol-mcp-server:latest
+        --port 4896
+
+In this example, the server is configured to use the generic OAuth Proxy provider.
+The server starts at the Docker container's port 4896, which is published to the
+host port 7766. Normally, the base URL of the MCP server will be a secure address
+of the server, as described in the next section. For example, if the server is
+deployed in an AWS EC2 instance, the EXA_AUTH_BASE_URL can be set to
+"https://ec2-xx-xxx-xx-xx.my-region.compute.amazonaws.com".
+
+Note that the path to the settings.json file must be an absolute path.
+
+Reverse proxy configuration
+---------------------------
+
+Exasol MCP Server runs as an HTTP server. Connecting to the server via an unsecured
```

Comment:
Is there a way to configure a SSL certificate such that the mcp server itself speaks https? If you don't know yet, this is probably something we should find out in the next cycle.